extensions/integration-tests/src/utils.ts

```diff
@@ -105,13 +104,33 @@ export async function deleteDB(dbName: string, ownerUri: string): Promise<void>
 			SELECT ERROR_MESSAGE() AS ErrorMessage;
 		END CATCH`;
 
-	await queryProvider.runQueryAndReturn(ownerUri, query);
+	ownerUri = await ensureServerConnected(server, ownerUri);
+	let dbDeleteResult = await this.runQuery(query, ownerUri);
+	assert(dbDeleteResult.columnInfo[0].columnName !== 'ErrorMessage', 'DB deletion threw error');
 }
 
+export async function ensureServerConnected(server: TestServerProfile, ownerUri: string): Promise<string> {
+	let connection = await azdata.connection.getConnection(ownerUri);
+	if (isNullOrUndefined(connection)) {
+		//create new connection if not present and use OwnerUri from there
+		let connectionId = await connectToServer(server);
+		return azdata.connection.getUriForConnection(connectionId);
+	}
+	return ownerUri;
+}
+
+
 export async function runQuery(query: string, ownerUri: string): Promise<azdata.SimpleExecuteResult> {
-	let queryProvider = azdata.dataprotocol.getProvider<azdata.QueryProvider>('MSSQL', azdata.DataProviderType.QueryProvider);
-	let result = await queryProvider.runQueryAndReturn(ownerUri, query);
-	return result;
+	try {
+		let queryProvider = azdata.dataprotocol.getProvider<azdata.QueryProvider>('MSSQL', azdata.DataProviderType.QueryProvider);
+		let result = await queryProvider.runQueryAndReturn(ownerUri, query);
+		return result;
+	}
+	catch (ex) {
+		console.error(ex);
```

Comment:
It's likely this isn't going to just be a string so you should check the type and extract the message if it's an Error object

Reply:
console.error should be able to log as is, and I want to log wole thing.

Reply:
If it's an object it'll be ok in the console but then just turn into basically useless text like [Object] when printed to a log.

https://typescript-play.js.org/#code/FAYw9gdgzmA2CmA6WYDmAKA3gAgLbyigENV5sAubAcgAl5YUBCKgXwEoBuIA

JSON.stringify is usually how we print out objects to the console (and that can handle strings just fine)